lib/crowbar/client/command/proposal.rb

```diff
@@ -101,33 +106,160 @@ module Proposal
             parent.desc "Create a proposal for specific barclamp"
             parent.arg :barclamp
             parent.arg :proposal
-            parent.command :create do |c|
+            parent.command "create" do |c|
+              c.desc "Proposal data in json format"
+              c.flag [:data], type: String, default_value: nil
+
+              c.desc "Proposal data as a json file"
+              c.flag [:file], type: String, default_value: nil
+
+              c.desc "Merge input with proposal data"
+              c.switch [:merge], negatable: false
+
               c.action do |global, opts, args|
                 barclamp = args.shift
                 proposal = args.shift
                 helper.validate_availability_of! barclamp
 
-                raise "Not implemented yet! (#{barclamp}, #{proposal})"
+                template = Request.instance.proposal_template(
+                  barclamp
+                ).parsed_response.easy_merge(
+                  "id" => proposal
+                )
+
+                case
+                when opts[:data]
+                  json = begin
+                    JSON.load opts[:data]
+                  rescue
+                    exit_now! "Invalid json data"
+                  end
+
+                  payload = if opts[:merge]
+                    template.easy_merge json
+                  else
+                    json
+                  end
+                when opts[:file]
+                  json = begin
+                    JSON.load opts[:file]
+                  rescue
+                    exit_now! "Invalid json file"
+                  end
+
+                  payload = if opts[:merge]
+                    template.easy_merge json
+                  else
+                    json
+                  end
+                else
+                  begin
+                    editor = Util::Editor.new content: template
+                    editor.edit!
+
+                    payload = editor.result
+                  rescue EditorAbortError => e
+                    exit_now! e.message
+                  rescue EditorStartupError => e
+                    exit_now! e.message
+                  rescue InvalidJsonError => e
+                    exit_now! e.message
+                  rescue
+                    exit_now! "Editing content failed"
+                  end
+                end
+
+                Request.instance.proposal_create(barclamp, proposal, payload) do |request|
+                  case request.code
+                  when 200
+                    say "Created successfully #{proposal} on #{barclamp}"
```

Comment:
"Successfully created..." sounds better to my ears. (same for all other strings)